### PR TITLE
Update to use Redis by default

### LINKED ISSUE #383: Change default persistence to Redis
Update scripts to default to Redis

### PR DESCRIPTION
**Pull request by andresrinivasan**
Fixes #383.

Tested via blackbox tests using the 1.1.0 Docker containers

Also

- _Chmod'ed +x_ a couple of files in `bin` that were missing the execute bit
- Tweaked `bin/run.sh` so that it is independent of `$(cwd)`
- Added support for referencing a Docker Compose file that was on the file system
- Updated docs, tried to simplify some of the steps, removed requirement to edit scripts

### CODE REVIEW

**Comment by jinlinGuan** [comment]
There are some relevant changes in [edgexfoundry/blackbox-testing/374](https://github.com/edgexfoundry/blackbox-testing/pull/374)

File: sync.sh [hunk 1]
```diff
@@ -7,14 +7,18 @@
 [ "$SECURITY_SERVICE_NEEDED" != true ] && USE_NO_SECURITY="-no-secty"
 
 # redis or mongo
-[ "$FOR_REDIS" = true ] && USE_REDIS="-redis"
+if [ "${FOR_REDIS:=true}" = true ]; then 
+     PERSIST="-redis"
+else
+     PERSIST="-mongo"
```
Comment (jinlinGuan) [inline sync.sh:13]: The names of the compose files in developer-scripts don't contain "mongo".

File: bin/run.sh [hunk 2]
```diff
@@ -8,15 +8,10 @@ fi
 
 # Ensure we fail the job if any steps fail
 set -e -o pipefail
-#. $(dirname "$0")/env.sh
 
 # Run the compose file for blackbox testing
 export network=$(docker network ls | awk '{print $2}' | grep edgex-network)
-docker-compose -f ../docker-compose-test-tools.yml up -d app-service-configurable
-
-option="${1}"
```
Comment (jinlinGuan) [inline bin/run.sh:17]: `option` is needed for the switch case.

File: bin/run.sh [hunk 3]
```diff
@@ -8,15 +8,10 @@ fi
...
-
-echo " ${option}"
+docker-compose -f "$(dirname "$0")"/../docker-compose-test-tools.yml up -d app-service-configurable
```
Comment (jinlinGuan) [inline bin/run.sh:16]: The `$(dirname "$0")` is `bin`, which causes ERROR here. The `docker-compose-test-tools.yml` is under the parent directory.

**Review by jinlinGuan — CHANGES_REQUESTED** [review]

File: sync.sh (hunk 1 above)
Comment (andresrinivasan) [reply sync.sh:13]: Please see https://github.com/edgexfoundry/developer-scripts/pull/218

Comment (jinlinGuan) [reply sync.sh:13]: Then the compose files with security services (docker-compose-nexus.yml and docker-compose-nexus-arm64.yml) need to contain "mongo" as well.

Comment (jinlinGuan) [reply sync.sh:13]: PR [edgexfoundry/developer-scripts#223](https://github.com/edgexfoundry/developer-scripts/pull/223) fixed it

**Comment by jinlinGuan** [comment]
@andresrinivasan This PR need to be merged ASAP in order to run tests on Jenkins properly, since [edgexfoundry/developer-scripts#218](https://github.com/edgexfoundry/developer-scripts/pull/218) has been merged. Thanks.

File: bin/run.sh (hunk 2 above)
Comment (andresrinivasan) [reply bin/run.sh:17]: Apologies for the fat finger. Corrected.

File: bin/run.sh (hunk 3 above)
Comment (andresrinivasan) [reply bin/run.sh:16]: What should be happening is `bin/run.sh` and `cd bin && ./run.sh` should now both work. In the former case `$(dirname $0)` will be `bin` so `bin/../docker-....yml` should be found. In the latter case, `$(dirname $0)` will be `.` so `./../docker-....yml` should be found. I'm seeing this in my testing.

How do I look at the failure in the Jenkins build?

Comment (andresrinivasan) [reply bin/run.sh:16]: The issue was not line 14 but line 147. The `updateMaxResultCount.sh` script didn't have an exit statement and the `set -e` on line 12 was triggered. This is corrected now.

File: bin/run.sh [hunk 4]
```diff
@@ -134,7 +131,6 @@ testAll() {
   coreDataTest
   commandTest
   metaDataTest
-  loggingTest
```
Comment (jinlinGuan) [inline bin/run.sh:137]: Is that because Redis doesn't need `loggingTest`? Mongo still needs it, so I don't think  it should be removed now.

Comment (cloudxxx8) [reply bin/run.sh:137]: @andresrinivasan although we run logging with file as persistence, we still need to run logging test to make sure the API works.
Theoretically, logging service APIs can work with file

File: bin/run.sh (hunk 3 above)
Comment (jinlinGuan) [reply bin/run.sh:16]: When executing `bin/run.sh -cd`, I got the following ERROR:
image
In this case, `$(dirname $0)` should be added to all the test scripts.

`cd bin && ./run.sh -cd`works fine.

**Review by cloudxxx8 — APPROVED** [review]
This PR is mostly good enough.
However, the blackbox test is broken and need this PR to be merged as soon as possible.
I am merging it and Ginny will open other issues to fix the error in this PR.

File: bin/run.sh (hunk 3 above)
Comment (andresrinivasan) [reply bin/run.sh:16]: Ah. I see what you're seeing now. Opening another issue to track.

**Pull request merged**